### ISSUE TITLE
Unsupported engine version in GKE Deployment template

### ISSUE THREAD

**Issue by iblazhko** [istio/istio#6520]
**Describe the bug**
GKE Deployment template uses unsupported version of GKE engine (`1.9.7-gke.1`) which causes GKE Deployment to fail.

**Expected behavior**
Following Quick start guide should not result in an error.

**Steps to reproduce the bug**
1. Follow instructions in https://istio.io/docs/setup/kubernetes/quick-start-gke-dm/
1. Section https://istio.io/docs/setup/kubernetes/quick-start-gke-dm/#launch-deployment-manager step 1 launches GKE Deployment manager with template <https://raw.githubusercontent.com/istio/istio/release-0.8/install/gcp/deployment_manager/istio-cluster.jinja>. This template uses GKE engine version `1.9.7-gke.1`

Deployment fails with the error
```master version "1.9.7-gke.1" is unsupported```

Equivalent command-line option `gcloud container clusters create istio-quick-start --cluster-version=1.9.7-gke.1 --zone europe-west4-b --num-nodes 3` fails with the same error.

Changing GKE version to `1.9.7-gke.3` solves the problem, so it looks like the template used by the QuickStart needs to be updated.

**Version**
Istio 0.8.0

**Is Istio Auth enabled or not?**
MTLS is disabled, RBAC is disabled

**Environment**
Google Cloud

**Comment by apettey** [comment]
@ymesika I ran into this issue today, I pulled and configured for 1.10.4-gke.2. What do you think about upgrading the PR to match with latest? Perhaps it will be supported for longer than 1.9.7-gke.3, although the rate goggle is going this might be void.

**Comment by iblazhko** [comment]
@ymesika I don't think there is such thing as a LTS GKE, see https://cloud.google.com/kubernetes-engine/versioning-and-upgrades

**Comment by apettey** [comment]
@ymesika Some people are saying that 1.10 is not supported, I found that the following two bugs have been resolved taken from the notes:
```
# NOTE: Due to a bug in kubectl (https://github.com/kubernetes/kubectl/issues/384 
# https://github.com/istio/issues/issues/261) we have to force an earlier
# version of kubectl than 1.10.0. 
```
I am spinning up another GKE cluster using `latest`, it did work so I will test and give feedback here.

**Comment by ymesika** [comment]
Thanks @apettey.
I meant that maybe we specify just a major version instead of being very picky about the exact one.
I believe that minimum requirement for 1.0 will be 1.9 so we can just use that tag.

**Comment by apettey** [comment]
To finalize, everything works great right `now` using latest instead of `1.9.7-gke.3` and `1.10.4-gke.2`. General tests and everything is running, this is still using kubectl 1.9.6.

My view is that latest can be considered a bit dangerous however with the current upgrade paths of Kubernetes and GKE it is more than likely the best solution right now. Short term it will reduce the number of PR's for each version of GKE that is deprecated.

**Comment by ymesika** [comment]
With #6909 we will get in 1.0 the latest available 1.9.
